src/graft_dag.cpp

```diff
@@ -0,0 +1,524 @@
+// Copyright 2019-2022 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+
+#include "graft_dag.hpp"
+
+#include "gp_dag.hpp"
+#include "quartet_hybrid_request.hpp"
+#include "reindexer.hpp"
+#include "subsplit_dag.hpp"
+
+// ** Constructors
+
+GraftDAG::GraftDAG(SubsplitDAG &dag) : host_dag_(dag) {}
+
+// ** Comparators
+
+int GraftDAG::Compare(GraftDAG &dag_a, GraftDAG &dag_b) {
+  SubsplitDAG &host_a = dag_a.GetHostDAG();
+  SubsplitDAG &host_b = dag_b.GetHostDAG();
+  // (1) Compare host DAGs.
+  int host_diff = SubsplitDAG::Compare(host_a, host_b);
+  if (host_diff != 0) {
+    return host_diff;
+  }
+  // (2) Compare graft nodes.
+  BitsetVector nodes_a = dag_a.GetSortedVectorOfNodeBitsets();
+  BitsetVector nodes_b = dag_b.GetSortedVectorOfNodeBitsets();
+  for (size_t i = 0; i < std::min(nodes_a.size(), nodes_b.size()); i++) {
+    if (Bitset::Compare(nodes_a[i], nodes_b[i]) != 0) {
+      return Bitset::Compare(nodes_a[i], nodes_b[i]);
+    }
+  }
+  if (nodes_a.size() - nodes_b.size() != 0) {
+    return nodes_a.size() - nodes_b.size();
+  }
+  // (3) Compare graft edges.
+  BitsetVector edges_a = dag_a.GetSortedVectorOfEdgeBitsets();
+  BitsetVector edges_b = dag_b.GetSortedVectorOfEdgeBitsets();
+  for (size_t i = 0; i < std::min(edges_a.size(), edges_b.size()); i++) {
+    if (Bitset::Compare(edges_a[i], edges_b[i]) != 0) {
+      return Bitset::Compare(edges_a[i], edges_b[i]);
+    }
+  }
+  if (edges_a.size() - edges_b.size() != 0) {
+    return edges_a.size() - edges_b.size();
+  }
+  return 0;
+}
+
+std::pair<BitsetVector, BitsetVector> BitsetVectorDiff(BitsetVector &vector_a,
```

Comment:
I don't see this being used, nor being tested.

Reply:
Removed.